src/dosext/mfs/lfn.c

```diff
@@ -1342,3 +1342,75 @@ int mfs_lfn(void)
 		CARRY;
 	return ret;
 }
+
+static int mfs_remap_extended_open_(void)
+{
+  char *src = MK_FP32(_DS, _SI);
+  uint8_t dosint;
+  int drive, found;
+  char fpath[PATH_MAX];
+  struct stat st;
+
+  d_printf("SFN: extended open '%s'\n", src);
+
+  drive = build_posix_path(fpath, src, 0);
+  if (drive < 0)
+    return 0;
+
+  found = find_file(fpath, &st, drive, NULL);
+  d_printf("SFN: extended open '%s' does %sexist\n", fpath, found ? "" : "not ");
+
+  if ((_DX & 0b00000011) && !found)   // fail if doesn't exist
+    return lfn_error(FILE_NOT_FOUND); // 0b01 open // 0b10 truncate
+
+  if ((_DX & 0b00010000) && found)    // fail create if exists
+    return lfn_error(FILE_ALREADY_EXISTS);
+
+  dosint = (_AL == 1) ? 0x3c : 0x3d;  // select create / open function
+
+  /* Map the extended open into something the underlying DOS can understand */
+  /*
+     low byte of BX (only bit 2 lost)
+       bit 2 read-only no update access time
+
+     high byte of BX (all bits lost)
+       bit 8 disable buffering
+       bit 9 disable file compression
+       bit 10 use _DI value for file name mangling
+       bit 13 return error instead of int 24
+       bit 14 synchronous writes
+   */
+  _AL = (_BL & 0b11110011);
+
+  /* Filename passed in DS:DX */
+  _DX = _SI;
+
+  call_dos_helper(dosint);
```

Comment:
I think this is not correct.
This helper works with `LFN_short_name`
which you don't seem to be filling with
short name. You need `call_msdos()`
API, plus I don't think it will then belong
to `lfn.c` at all (so can be as well coded
in `int.c`).

Reply:
Besides, `call_dos_helper()` only prepares
the call on stack. It returns before call even
started. So messing with CF and return values
is wrong. You need `call_msdos()` - it actually
does the call.

Reply:
Mmm, I see what you mean about call_dos_helper(). I guess it works in my test because I'm going from LFN 716c - > SFN 6c00 -> SFN 3d00. If an app just called SFN 6c00 then who knows what file SFN 3d00 would try to open.

I'll have a rethink, thanks for the review.